src/Server/Query.elm

```diff
@@ -748,28 +749,69 @@ maybeMakingWasteParser key =
             )
 
 
-maybeYarnSize : String -> Parser (ParseResult (Maybe Unit.YarnSize))
-maybeYarnSize key =
-    Query.int key
+parseYarnSize : String -> Maybe Unit.YarnSize
+parseYarnSize str =
+    let
+        withUnitRegex =
+            -- Match either an int or a int and a unit `Nm` or `Dtex`
+            Regex.fromString "(\\d+)\\s?(Nm|Dtex)"
```

Comment:
I'd disallow the optional space between the number and the unit, especially as a query string parameter value

Reply:
good point, thanks